### PR TITLE
[DOC-325] [2.21.1] Added ycql-stat-statements and yb_local_tablets

### PR DESCRIPTION
**Pull request by aishwarya24**

### CODE REVIEW

File: docs/content/preview/explore/observability/yb-local-tablets.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+---
+title: View YSQL/YCQL and tablet metadata with yb_local_tablets
+linkTitle: YQL and tablet metadata
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/yb-local-tablets.md:3]: is yql needed here?

File: docs/content/preview/explore/observability/yb-local-tablets.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+| Column | Type | Description |
+| :----- | :--- | :---------- |
+| tablet_id | text | 16 byte UUID of the tablet. |
+| table_id | text | 16 byte UUID of the table which the tablet is part of. |
+| table_type | text | Type of the table. Can be YSQL, YCQL, System, or Unknown. |
+| namespace_name | text | Name of the database or the keyspace. |
+| ysql_schema_name | text | YSQL schema name. Empty for YCQL, System, and Unknown table types. |
+| table_name | text | Name of the table which the tablet is part of. |
+| partition_key_start| bytea | Start key of the partition (inclusive). |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/yb-local-tablets.md:28]: Can we add here that the value will be NULL if it's the first partition?

File: docs/content/preview/explore/observability/yb-local-tablets.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+| partition_key_end  | bytea | End key of the partition (exclusive).|
```
Comment (abhinab-yb) [inline docs/content/preview/explore/observability/yb-local-tablets.md:29]: and here, the value will be NULL if it's the last partition

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+This view is added in an extension `yb_ycql_utils`, which is not enabled by default.
+
+The columns of the `ycql_stat_statements` view are described in the following table.
+
+| Column | Type | Description |
+| :----- | :--- | :---------- |
+| queryid | int8 | Hash code to identify identical normalized queries. |
+| query | text | Text of a representative statement. |
+| is_prepared  | bool | Indicates whether the statement is a prepared statement or an unprepared query. |
```
Comment (abhinab-yb) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:31]:
```suggestion
| is_prepared  | bool | Indicates whether the statement is a prepared an unprepared statement. |
```

**Review by abhinab-yb — APPROVED** [review]

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
+---
+title: Get query statistics using ycql_stat_statements
+linkTitle: Get query statistics
+description: Track planning and execution statistics for all YCQL statements executed by a server.
```
Comment (hbhanawat) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:4]: It tracks only execution statistics. Please change at other places as well.

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    weight: 200
+type: docs
+---
+
+{{<tabs>}}
+{{<tabitem href="../pg-stat-statements/" text="YSQL" icon="postgres" >}}
+{{<tabitem href="../ycql-stat-statements/" text="YCQL" icon="cassandra" active="true" >}}
+{{</tabs>}}
+
+Databases can be resource-intensive, consuming a lot of memory CPU, IO, and network resources. Optimizing your CQL can be very helpful in minimizing resource utilization. The `ycql_stat_statements` module helps you track planning and execution statistics for all the YCQL statements executed by a server.
```
Comment (hbhanawat) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:19]: Tracks only execution statistics. Change at other places as well.

File: docs/content/preview/explore/query-1-performance/ycql-stat-statements.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+This view provides YCQL statement metrics (similar to pg_stat_statements) that are also present on `<yb-tserver-ip>:12000/statements`, accessible via YSQL. The view can be joined with YCQL wait events in the [yb_active_session_history](../../observability/active-session-history/#yb-active-session-history) view using the query ID.
```
Comment (hbhanawat) [inline docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:21]: This view provides YCQL statement metrics (similar to pg_stat_statements) that are also present on `<yb-tserver-ip>:12000/statements`, accessible via YSQL

Please remove ", accessible via YSQL. It is not making sense. This is a YCQL endpoint only.

Comment (aishwarya24) [reply docs/content/preview/explore/query-1-performance/ycql-stat-statements.md:21]: Oh I added it based on this response : https://docs.google.com/document/d/1VRiPYWTclpxBnyxxix9yD87WIZQrsALviDY5qZXha9Y/edit?disco=AAABMAeLnqA
@hbhanawat

File: docs/content/preview/explore/observability/yb-local-tablets.md (hunk 1 above)
Comment (aishwarya24) [reply docs/content/preview/explore/observability/yb-local-tablets.md:3]: changing to view tablet metadata.

**Comment by aishwarya24** [comment]
Created another one - https://github.com/yugabyte/yugabyte-db/pull/22564

**Pull request closed without merging**